Source/Core/Core/IOS/Network/KD/NetKDRequest.cpp

```diff
@@ -211,6 +306,96 @@ void NetKDRequestDevice::LogError(ErrorType error_type, s32 error_code)
   m_scheduler_buffer[2] = Common::swap32(new_code);
 }
 
+NWC24::ErrorCode NetKDRequestDevice::KDCheckMail(u32* mail_flag, u32* interval)
+{
+  Common::ScopeGuard state_guard([&] {
+    std::lock_guard lg(m_scheduler_buffer_lock);
+    m_scheduler_buffer[4] = static_cast<u32>(CurrentFunction::None);
+  });
+
+  std::unique_lock lg(m_scheduler_buffer_lock);
+  m_scheduler_buffer[4] = Common::swap32(static_cast<u32>(CurrentFunction::Check));
+  lg.unlock();
```

Comment:
Okay... I understand what you tried to do here but this is just messy. Please just make an extra scope, so

```
{
  std::lock_guard lg(m_scheduler_buffer_lock);
  m_scheduler_buffer[4] = Common::swap32(static_cast<u32>(CurrentFunction::Check));
}
```

For the ScopeGuard, You're now quickly locking and unlocking the mutex twice on success, which is not ideal. I'd recommend:

```
  bool success = false;
  Common::ScopeGuard state_guard([&] {
    std::lock_guard lg(m_scheduler_buffer_lock);
    if (success)
    {
      // m_scheduler_buffer[11] contains the amount of times we have checked for mail this IOS session.
      m_scheduler_buffer[11] = Common::swap32(Common::swap32(m_scheduler_buffer[11]) + 1);
    }
    m_scheduler_buffer[4] = static_cast<u32>(CurrentFunction::None);
  });
```

And then set `success = true;` at the bottom instead of the write to `m_scheduler_buffer[11]`.